src/Features/CSharp/Portable/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.Rewriting.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertIfToSwitch
+{
+    using static SyntaxFactory;
+
+    internal sealed partial class CSharpConvertIfToSwitchCodeRefactoringProvider
+    {
+        public override SyntaxNode CreateSwitchExpressionStatement(SyntaxNode target, ImmutableArray<AnalyzedSwitchSection> sections)
+        {
+            return ReturnStatement(
+                SwitchExpression((ExpressionSyntax)target, SeparatedList(sections.Select(AsSwitchExpressionArmSyntax))));
+        }
+
+        private static SwitchExpressionArmSyntax AsSwitchExpressionArmSyntax(AnalyzedSwitchSection section)
+        {
+            return SwitchExpressionArm(
+                pattern: section.Labels.IsDefault
+                    ? DiscardPattern()
+                    : AsPatternSyntax(section.Labels[0].Pattern),
+                whenClause: section.Labels.IsDefault // TODO
+                    ? null
+                    : AsWhenClause(section.Labels[0]),
```

Comment:
why only the first label in both these cases?

Reply:
that must be the only label since we're generating a switch expr. will add an assert and comment.